osgar/tools/repare_log.py

```diff
@@ -0,0 +1,35 @@
+"""
+  Utility for repairing log files with missing header
+"""
+import sys
+import os
+from osgar.logger import LogWriter
+
+
+def fix_log_header(filename):
+
+    with open(filename, 'rb') as f:
+        data = f.read()
+    if data.startswith(b'Pyr'):
+        return  # seems to be OK
+
+    start_index = data.index(b'cmd_vel') - 2 - 8  # msgpack index and IHH time, stream, len
+    end_index = data.rindex(b'cmd_vel') - 2 - 8  # msgpack index and IHH time, stream, len
+
+    new_filename = filename + '.new'
+    with LogWriter(filename=new_filename) as log:
+        for i in range(1, 25):  # TODO check count
+            log.register('name'+str(i))
+        log.f.write(data[start_index:end_index])
```

Comment:
I don't understand what is being done in this function. Why would b`cmd_vel` be part of the data? What kind of log is it "repairing"? This does not look like the generic repair we have been discussing.

Reply:
it is specific for Virtual - I am dropping it for now (it should disappear after rebase)